src/bot/commands/command-play.ts

```diff
@@ -45,23 +25,47 @@ export class Play extends Command {
         guildId: String(voiceMember.guild.id),
       });
 
-      const player = this.getPlayer(voiceMember.channelId);
-      connection.subscribe(player);
+      const source = this.getSourceStream();
 
       const queue = this.getQueue();
-      queue.add(voiceMember.channelId, {
-        audioResource,
-        streamInfo,
-        userSearch: input,
+
+      const video = await source.getStreamFromUrl(input);
+
+      const searchResult =
+        video ??
+        ((await source.search(input, { limit: 1 })) as Array<StreamInfo>);
+
+      searchResult.forEach((streamInfo: StreamInfo) => {
+        queue.add(voiceMember.channelId, {
+          streamInfo: {
+            title: streamInfo.title,
+            url: streamInfo.url,
+            artist: streamInfo.artist,
+          },
+          userSearch: input,
+        });
       });
 
-      let replyContent = `${message.author.username} ${BOT_MESSAGES.PUSHED_TO_QUEUE} ${streamInfo.title}`;
+      const player = this.getPlayer(voiceMember.channelId);
+      connection.subscribe(player);
+
+      let replyContent = `${message.author.username} ${BOT_MESSAGES.PUSHED_TO_QUEUE} ${searchResult[0].title} - ${searchResult[0].artist}`;
 
       if (player.state.status === AudioPlayerStatus.Idle) {
-        player.play(audioResource);
+        const searchResultUrl =
+          searchResult[0]?.url ??
```

Comment:
using array.shift() is better implementation then accessing by index
```suggestion
          const firstSong = searchResult.shift()
          //firstSong?.url?...
```

Reply:
resolved: [5252756](https://github.com/Marli-Marlete/marli-music/pull/21/commits/5252756b4ba49d970306720a3bb9a66014cda2c6)